test/dvotingdela.go

```diff
@@ -0,0 +1,519 @@
+	"github.com/stretchr/testify/require"
+	accessContract "go.dedis.ch/dela/contracts/access"
+	"go.dedis.ch/dela/contracts/value"
+	"go.dedis.ch/dela/core/access"
+	"go.dedis.ch/dela/core/access/darc"
+	"go.dedis.ch/dela/core/execution/native"
+	"go.dedis.ch/dela/core/ordering"
+	"go.dedis.ch/dela/core/ordering/cosipbft"
+	"go.dedis.ch/dela/core/ordering/cosipbft/authority"
+	"go.dedis.ch/dela/core/ordering/cosipbft/blockstore"
+	"go.dedis.ch/dela/core/ordering/cosipbft/types"
+	"go.dedis.ch/dela/core/store/hashtree"
+	"go.dedis.ch/dela/core/store/hashtree/binprefix"
+	"go.dedis.ch/dela/core/store/kv"
+	"go.dedis.ch/dela/core/txn"
+	"go.dedis.ch/dela/core/txn/pool"
+	poolimpl "go.dedis.ch/dela/core/txn/pool/gossip"
+	"go.dedis.ch/dela/core/txn/signed"
+	"go.dedis.ch/dela/core/validation"
+	"go.dedis.ch/dela/core/validation/simple"
+	"go.dedis.ch/dela/cosi/threshold"
+	"go.dedis.ch/dela/crypto"
+	"go.dedis.ch/dela/crypto/bls"
+	"go.dedis.ch/dela/crypto/loader"
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/mino/gossip"
+	"go.dedis.ch/dela/mino/minogrpc"
+	"go.dedis.ch/dela/mino/minogrpc/certs"
+	"go.dedis.ch/dela/mino/minogrpc/session"
+	"go.dedis.ch/dela/mino/router/tree"
+	"go.dedis.ch/dela/serde/json"
+	"golang.org/x/xerrors"
+)
+
+const certKeyName = "cert.key"
+const privateKeyFile = "private.key"
+
+var aKey = [32]byte{1}
+var valueAccessKey = [32]byte{2}
+
+// evotingAccessKey is the access key used for the evoting contract.
+var evotingAccessKey = [32]byte{3}
+
+// dela defines the common interface for a Dela node.
+type dela interface {
+	Setup(...dela)
+	GetMino() mino.Mino
+	GetOrdering() ordering.Service
+	GetTxManager() txn.Manager
+	GetAccessService() access.Service
+}
+
+// dVotingCosiDela defines the interface needed to use a Dela node using cosi.
+type dVotingCosiDela interface {
+	dela
+
+	GetAddr() net.Addr
+	GetPublicKey() crypto.PublicKey
+	GetPool() pool.Pool
+	GetAccessStore() accessstore
+	GetTree() hashtree.Tree
+	GetDkg() dkg.DKG
+	GetShuffle() shuffle.Shuffle
+
+	// jean ->
+	// adminID   string
+	// manager   *signed.TransactionManager
+	// firstNode dVotingNode
+	// actor     dkg.Actor
+	// signer    crypto.AggregateSigner
+	// <- jean
+}
+
+// dVotingNode represents a Dela node using cosi pbft aimed to execute d-voting
+//
+// - implements dela
+type dVotingNode struct {
+	t             *testing.T
+	onet          mino.Mino
+	ordering      ordering.Service
+	cosi          *threshold.Threshold
+	txManager     txn.Manager
+	addr          net.Addr
+	pool          pool.Pool
+	accessService access.Service
+	accessStore   accessstore
+	tree          hashtree.Tree
+	dkg           *pedersen.Pedersen
+	shuffle       *neff.NeffShuffle
+}
+
+// Creates n dela nodes using tempDir as root to file path and returns an array
+// of nodes or error
+func setupDVotingNodes(t *testing.T, numberOfNodes int, tempDir string) []dVotingCosiDela {
+	var dVotingNodes []dVotingCosiDela = make([]dVotingCosiDela, numberOfNodes)
```

Comment:
we never use the `var` to initialize. It is used only to declare a variable. There you should just do 

```go
dVotingNodes := make(...
```

Reply:
thanks